Tests/js/views/services/assets/ez-navigationhubviewservice-tests.js

```diff
@@ -406,6 +558,258 @@ YUI.add('ez-navigationhubviewservice-tests', function (Y) {
         },
     });
 
+    loadTest = new Y.Test.Case({
+        name: "eZ Navigation Hub View Service load test",
+
+        setUp: function () {
+            this.capiMock = new Mock();
+            this.discoveryServiceMock = new Mock();
+            this.contentRootMock = new Mock();
+            this.locationRootMock = new Mock();
+            this.contentMediaMock = new Mock();
+            this.locationMediaMock = new Mock();
+
+            this.service = new Y.eZ.NavigationHubViewService({
+                capi: this.capiMock,
+                rootStruct: {content: this.contentRootMock, location: this.locationRootMock},
+                rootMediaStruct: {content: this.contentMediaMock, location: this.locationMediaMock},
+            });
+        },
+
+        tearDown: function () {
+            this.service.destroy();
+            delete this.service;
+        },
+
+        _initDiscoveryService: function (fail) {
+            Mock.expect(this.capiMock, {
+                method: 'getDiscoveryService',
+                returns: this.discoveryServiceMock,
+            });
+
+            Y.Mock.expect(this.discoveryServiceMock, {
+                method: 'getInfoObject',
+                args: [Mock.Value.String, Y.Mock.Value.Function],
+                run: function (object, callback) {
+                    callback(fail ? true : false, {"_href": 'david-seaman'});
+                }
+            });
+        },
+
+        _initLocationMock: function (locationMock, fail) {
+            Mock.expect(locationMock, {
+                method: 'set',
+                args: [Mock.Value.String, Mock.Value.String],
```

Comment:
too weak, the first parameter is supposed to be `id` the second one is the id given by `getInfoObject`.